core/module.rbs

```diff
@@ -731,8 +731,8 @@ class Module < Object
   #     I'm Dino!
   #     #<B:0x401b39e8>
   #
-  def define_method: (interned symbol, ^() [self: self] -> untyped | Method | UnboundMethod method) -> Symbol
-                   | (interned symbol) { () [self: self] -> untyped } -> Symbol
+  def define_method: (interned symbol, ^() [self: top] -> untyped | Method | UnboundMethod method) -> Symbol
```

Comment:
We cannot have meaningful self type here because the type of `self` inside the block is the instance of modules that is an instance of `Module`. `instance` is `Module`, `self` is also `Module`.

Writing `top` here enforces writing type annotations for `self` -- `# @type self: T` in Steep.

Reply:
* It’ll have to be `singleton(self)` similar to #558.
* Or #1542.